### PR TITLE
Doc about Task, WIP

### PR DESCRIPTION
**Pull request by reactormonk**
Anything specific I should add? Got a few good ideas for examples?

### CODE REVIEW

File: docs/src/main/tut/task.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+menu: tut
+weight: 115
+title: Task[T]
+---
+
+As you might have noticed, a few methods, e.g. `withBody` lift a `Response` into a `Task` context, so you end up with a `Task[Response]`. Coincidentially, all the methods which relate to the body of the `Request`/`Response`. Why does this happen? http4s is built on [Streaming], so directly materializing the body of a `Response` defeats any streaming benefits. Instead, the result is wrapped in a `Task` which preserves the asynchronous nature.
```
Comment (rossabaker) [inline docs/src/main/tut/task.md]: Maybe s/coincidentally/namely/.  It's not a coincidence.

I'd say it's less about streaming and more about effects: we might know the content length before we start streaming bytes, but not until we stat the file or hear back from the database.  Both the streaming and the delayed content length calculation are about delaying side effects.

File: docs/src/main/tut/task.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## Methods on `Task[Response]`
+
+As with every other context, you have to map/flatMap it to do useful things with it. To avoid some of this, there's [Message], which puts quite a few methods on `Task[Request]`/`Task[Response]`. There's some more methods on [TaskRequestOps] and [TaskResponseOps].
```
Comment (rossabaker) [inline docs/src/main/tut/task.md]: It might be good to show an example here of what things would be like without this syntax, and then with.

**Comment by ChristopherDavenport** [comment]
In February this made a lot of sense, now I think it seems a bit more niche with #1197 on the horizon. Where do we stand on that state of this PR?

**Comment by reactormonk** [comment]
Drop it, link to cats-effect documentation?

**Comment by rossabaker** [comment]
It's probably still nice to say a few words about some operations appearing on both `Response` and `Task[Response]` (and likewise `Request`).  But a general discussion of effects should definitely refer to the cats-effect documentation.

With the deprecation of `fs2.Task`, I think `cats.effect.IO` will be what we see in our docs.

**Pull request closed without merging**